### PR TITLE
ignition support

### PR DESCRIPTION
**Pull request by mischief**

### CODE REVIEW

File: platform/conf.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+	defConfig = ign.Config{
+		Version: 1,
+	}
+)
+
+func NewConf(userdata string) (*Conf, error) {
+	c := &Conf{}
+
+	ignc, err := ign.Parse([]byte(userdata))
+	// empty, noop
```
Comment (vcaputo) [inline platform/conf.go]: any reason to not use a switch here instead?

**Comment by mischief** [comment]
unfortunately the test fails currently.

```
core@ip-172-31-5-138 ~ $ journalctl -t ignition
-- Logs begin at Tue 2015-10-27 04:34:04 UTC, end at Tue 2015-10-27 04:35:27 UTC. --
Oct 27 04:34:06 localhost ignition[363]: Ignition v0.2.2
Oct 27 04:34:06 localhost ignition[363]: op(1): [started]  GET "http://169.254.169.254/2009-04-04/user-data"
Oct 27 04:34:06 localhost ignition[363]: op(1): got data "{\"ignitionVersion\":1,\"storage\":{\"filesystems\":[{\"device\":\"/dev/disk/by-label/ROOT\",\"format\":\"ext4\",\"files\":[{\"path\"
Oct 27 04:34:06 localhost ignition[363]: op(1): [finished] GET "http://169.254.169.254/2009-04-04/user-data"
Oct 27 04:34:06 localhost ignition[363]: config successfully fetched
Oct 27 04:34:06 localhost ignition[363]: op(2): [started]  GET "http://169.254.169.254/2009-04-04/meta-data/public-keys"
Oct 27 04:34:06 localhost ignition[363]: op(2): got data "0=mischief"
Oct 27 04:34:06 localhost ignition[363]: op(2): [finished] GET "http://169.254.169.254/2009-04-04/meta-data/public-keys"
Oct 27 04:34:06 localhost ignition[363]: op(3): [started]  GET "http://169.254.169.254/2009-04-04/meta-data/public-keys/0/openssh-key"
Oct 27 04:34:06 localhost ignition[363]: op(3): got data "ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAw/9IepzQFsIPdImwymIeu7ya/Dxfx9CpKh/88ZnblJ0eLnYVQfAHNxYDCQXI8+KiZ7EDxfzewq2Hw0LQsAQ69Is/9AkCRK8gX
Oct 27 04:34:06 localhost ignition[363]: op(3): [finished] GET "http://169.254.169.254/2009-04-04/meta-data/public-keys/0/openssh-key"
Oct 27 04:34:06 localhost ignition[363]: found SSH public key for "0"
Oct 27 04:34:06 localhost ignition[363]: fetched config: {Version:1 Storage:{Disks:[] Arrays:[] Filesystems:[{Device:/dev/disk/by-label/ROOT Format:ext4 Create:<nil> Files:[{Path:/etc/hostham
Oct 27 04:34:06 localhost ignition[363]: disks: createFilesystems: op(1): [started]  waiting for devices [/dev/disk/by-label/ROOT]
Oct 27 04:34:06 localhost ignition[363]: disks: createFilesystems: op(1): [finished] waiting for devices [/dev/disk/by-label/ROOT]
Oct 27 04:34:06 localhost ignition[363]: disks: createFilesystemsFiles: createFiles: op(2): [started]  mounting "/dev/disk/by-label/ROOT" at "/tmp/ignition-files763800276"
Oct 27 04:34:06 localhost ignition[363]: disks: createFilesystemsFiles: createFiles: op(2): [finished] mounting "/dev/disk/by-label/ROOT" at "/tmp/ignition-files763800276"
Oct 27 04:34:06 localhost ignition[363]: disks: createFilesystemsFiles: createFiles: op(3): [started]  writing file "/etc/hosthame"
Oct 27 04:34:06 localhost ignition[363]: disks: createFilesystemsFiles: createFiles: op(3): [finished] writing file "/etc/hosthame"
Oct 27 04:34:06 localhost ignition[363]: disks: createFilesystemsFiles: createFiles: op(4): [started]  unmounting "/dev/disk/by-label/ROOT" at "/tmp/ignition-files763800276"
Oct 27 04:34:07 localhost ignition[363]: disks: createFilesystemsFiles: createFiles: op(4): [finished] unmounting "/dev/disk/by-label/ROOT" at "/tmp/ignition-files763800276"
Oct 27 04:34:08 localhost ignition[414]: Ignition v0.2.2
Oct 27 04:34:08 localhost ignition[414]: files: createUsers: op(1): [started]  adding ssh keys to user "core"
Oct 27 04:34:08 localhost ignition[414]: files: createUsers: op(1): [finished] adding ssh keys to user "core"
core@ip-172-31-5-138 ~ $ cat /etc/hostname                                                                                                                                                     
cat: /etc/hostname: No such file or directory
```

**Comment by vcaputo** [comment]
Path:/etc/hostham ?

**Comment by mischief** [comment]
ptal

**Comment by crawford** [comment]
lgtm

**Pull request merged**